W-13104940-API-console-proxy-for-PCE-fa (#315)
mulesoft/docs-private-cloud: modules/ROOT/pages/config-add-proxy-whitelist.adoc

include::_attributes.adoc[]
endif::[]

> == Registered Domains

Reviewer: This should be `Trusted Domains`

> Every HTTP request you make to an external resource goes through the API Console Proxy. For security reasons, you need to register the domain as an allowed or blocked domain in Anypoint Platform. 

Reviewer: In this case, we can only register domains as allowed (or trusted domains), but we can't register as blocked.  So I would remove the following: `or blocked domain`

> By default, proxy requests are blocked for new domains on authenticated requests.

Reviewer: I would change new domains for `external domains`.

To add a new domain:

1. Log in to Anypoint Platform using an account that has the Organization Administrator permission.

2. In the navigation bar or the main Anypoint Platform page, click *Access Management*.

> 3. Click *Registered domains*.

Reviewer: change for `3. Click *Trusted Domains*.`

4. Click *Add domain*. +

> The domain must not match with any existing domain, and must not be any domain or subdomain of MuleSoft or Anypoint Platform.

Reviewer: Also, we should specify that it must be a domain (for example, If I want to access https://mydomain.com I need to add `mydomain.com`). Also, there is no regular expressions support, like in the previous functionality. We should also add that.

> 5. Under Trust level, select if you want the domain to be *Allowed* or *Blocked*.

Reviewer: This belongs to the new UI which is not being onboarded. We should remove this step and go directly to the next one.

6. In the dialog box that appears, type the domain name and click Add domain. +

> Result: The domain has been added to your list of registered domains. You are redirected to the registered domains page, where you can edit or delete your domains. If you register a domain on a parent organization, it is also allowlisted on all your business groups.

Reviewer: We should change this whole paragraph for the following: `Result: The domain has been added to your trusted domains. You will be able to see it listed on the trusted domains page, where you can also delete your domains. If you register a domain on a parent organization, it will be also allowed on all your child business groups.`

include::_attributes.adoc[]
endif::[]

> By default, proxy requests are enabled for the domain name where the platform is running. 

Reviewer 2: ```suggestion
By default, proxy requests are enabled for the domain name where the platform runs. 
```

> To allow addresses:

Reviewer 2: ```suggestion
To allow addresses in versions 3.2.0 and 3.2.1:
```

To add a new domain:

1. Log in to Anypoint Platform using an account that has the Organization Administrator permission.

2. In the navigation bar or the main Anypoint Platform page, click *Access Management*.

3. Click *Trusted domains*.

4. Click *Add domain*. +
The domain must not match with any existing domain,  must not be any domain or subdomain of MuleSoft or Anypoint Platform, and should be written in the format `exampledomain.com`.
NOTE: Regular expressions are not supported at the moment.

5. In the dialog box that appears, type the domain name and click Add domain. +

Result: The domain has been added to your list of trusted domains. You are redirected to the trusted domains page, where you can edit or delete your domains. If you register a domain on a parent organization, it is also allowlisted on all your business groups.

== Authentication 

When using the proxy, there are two authentication modes available: 

*Strict Authentication*: All requests reaching the proxy must be authenticated. This means that user token credentials must be included in the ms-authorization header of the request. This is the default option.

> WARNING: Requests to external domains from the Exchange Public Portals will fail, as there is no Anypoint user involved in the event, and external domains require authentication.

Reviewer: WARNING: As requests to external domains from the Exchange Public Portals go through API Console Proxy, those requests will fail as there is no Anypoint user involved in the call.

> *Non-Strict Authentication*: ACP continues to be unauthenticated, but also allows unauthenticated requests. The incoming request must provide either the users credentials or a reference ot the Organization ID that the request belongs to.

Reviewer: API Console Proxy continues to be authenticated but also allows unauthenticated requests. The incoming request must provide either the user's token or a reference to the Organization ID that the request belongs to.  The Organization ID must be sent on the `ms-organization-id` header in the call.

To change the authentication method from strict to non-strict, follow these steps:

. Log in to Ops Center and select *Kubernetes* on the left side-bar.

> . Select the `api-console-proxy` namespace from the drop-down menu on the top, and then select *Configmaps*. Find the xservice-env` configmap and then click on *Edit Config Map*.

Reviewer: Find the `service-env` configmap

----
kubectl delete pod -n api-manager -l component=api-platform-web
----

> NOTE: This method is currently working from version 3.2.0 to version 3.2.1. From version 3.2.2 and any future versions, refer to Trusted Domains.

Reviewer 2: ```suggestion
For versions 3.2.2 and later, refer to Trusted Domains.
```

Reviewer 2: Does "This method" refer to the last step or to the whole procedure? It would be better to have this information before the procedure.

> NOTE: This method is currently working from version 3.2.0 to version 3.2.1. From version 3.2.2 and any future versions, refer to Trusted Domains.

Reviewer 2: ```suggestion
NOTE: This method works for versions 3.2.0 to 3.2.1. For version 3.2.2 and later, refer to Trusted Domains.
```

> NOTE: This method is currently working from version 3.2.0 to version 3.2.1. From version 3.2.2 and any future versions, refer to Trusted Domains.

Reviewer 2: It is not clear what "this method" refers to. Is it all of the preceding steps or just the last one? If it refers to all of the preceding steps, it would be better to mention in the lead-in to the procedure that the method works for Versions 3.2.0 and 3.2.1.

== Trusted Domains

> Every HTTP request you make to an external resource goes through the API Console Proxy. For security reasons, you need to register the domain as an allowed domain in Anypoint Platform. 

Reviewer 2: ```suggestion
Every HTTP request you make to an external resource goes through the API Console Proxy. For security reasons, you must register the domain as an allowed domain in Anypoint Platform. 
```

By default, proxy requests are blocked for external domains on authenticated requests.

To add a new domain:

1. Log in to Anypoint Platform using an account that has the Organization Administrator permission.

> 2. In the navigation bar or the main Anypoint Platform page, click *Access Management*.

Reviewer 2: For step 2, pick one method to document.

3. Click *Trusted domains*.

> 4. Click *Add domain*. +

Reviewer 2: ```suggestion
4. Click *Add domain*. 
+
```

Reviewer 2: The plus sign should be on its own line to leave a vertical space.

> The domain must not match with any existing domain,  must not be any domain or subdomain of MuleSoft or Anypoint Platform, and should be written in the format `exampledomain.com`.

Reviewer 2: ```suggestion
The domain must not match an existing domain, must not be a domain or subdomain of MuleSoft or Anypoint Platform, and must be written in the format `exampledomain.com`.
```

> NOTE: Regular expressions are not supported at the moment.

Reviewer 2: ```suggestion
NOTE: Regular expressions are not supported.
```

> 5. In the dialog box that appears, type the domain name and click Add domain. +

Reviewer 2: ```suggestion
5. Type the domain name and click *Add domain*.
```

> Result: The domain has been added to your list of trusted domains. You are redirected to the trusted domains page, where you can edit or delete your domains. If you register a domain on a parent organization, it is also allowlisted on all your business groups.

Reviewer 2: ```suggestion
The domain is added to your list of trusted domains. You are redirected to the trusted domains page, where you can edit or delete your domains. If you register a domain on a parent organization, the domain is allowlisted for all of your business groups.
```

== Authentication 

> When using the proxy, there are two authentication modes available: 

Reviewer 2: ```suggestion
When using the proxy, there are two authentication modes: 
```

> *Strict Authentication*: All requests reaching the proxy must be authenticated. This means that user token credentials must be included in the ms-authorization header of the request. This is the default option.

Reviewer 2: ```suggestion
*Strict Authentication*
+
(Default) All requests that reach the proxy must be authenticated, which means that user token credentials must be included in the ms-authorization header of the request.
```

Reviewer 2: To save words, I suggest mentioning the default like this:

*Strict Authentication*
+
(Default) All requests that reach...

> WARNING: As requests to external domains from the Exchange Public Portals go through the API Console Proxy, they will fail as no Anypoint user is involved in the call.

Reviewer 2: ```suggestion
WARNING: If requests to external domains from the Exchange Public Portals go through the API Console Proxy, they fail because no Anypoint user is involved in the call.
```

> *Non-Strict Authentication*: API Console Proxy continues to be unauthenticated, but also allows unauthenticated requests. The incoming request must provide either the user's token or a reference to the Organization ID that the request belongs to. The Organization ID must be sent on the `ms-organization-id` header.

Reviewer 2: ```suggestion
*Non-Strict Authentication*
+
API Console Proxy continues to be unauthenticated but it allows unauthenticated requests. The incoming request must provide either the user's token or a reference to the Organization ID that the request belongs to. The Organization ID must be sent on the `ms-organization-id` header.
```

To change the authentication method from strict to non-strict, follow these steps:

. Log in to Ops Center and select *Kubernetes* on the left side-bar.

> . Select the `api-console-proxy` namespace from the drop-down menu on the top, and then select *Configmaps*. Find the `service-env` configmap and then click on *Edit Config Map*.

Reviewer 2: ```suggestion
. Select the `api-console-proxy` namespace from the drop-down menu on the top, and select *Configmaps*. Locate the `service-env` configmap and click on *Edit Config Map*.
```

Reviewer 2: I think that the sentence that starts with "Locate" should be a separate step.

> . Select the *STRICT_AUTHENTICATION_FF* tab and ensure that the value is set to `false`. (You'll have to hover your mouse over the tabs to get a tooltip with the full name to find the right tab)

Reviewer 2: ```suggestion
. Select the *STRICT_AUTHENTICATION_FF* tab and ensure that the value is set to `false`. (To find the right tab, hover your mouse over the tabs to get a tooltip with the full name.)
```

Reviewer 2: Regarding the edit above, it is best to start with the why before saying the how. That way, the user knows whether he/she needs to read the rest of the sentence.

> . Select *Apply* to save changes to the `service-env` config map.

Reviewer 2: Select > Click (I assume *Apply* is a button.)

> . Re-create the pods to ensure that each node in the cluster uses the most recent configuration:

Reviewer 2: Our style guide says not to use a hyphen for recreate.
